util-core/src/main/scala/com/twitter/io/Reader.scala

```diff
@@ -157,8 +157,8 @@ object Reader {
     require(chunkSize > 0, s"chunkSize should be > 0 but was $chunkSize")
 
     @tailrec
-    private def loop(acc: Seq[Buf], in: Buf): Seq[Buf] = {
-      if (in.length < chunkSize) acc :+ in
+    private def loop(acc: ListBuffer[Buf], in: Buf): Seq[Buf] = {
```

Comment:
this isn't getting any prettier. I'd be happy to replace with a non-mutable loop, but went for minimal invasiveness